src/ensembl/src/content/app/browser/track-panel/track-panel-modal/modal-views/TrackPanelBookmarks.test.tsx

```diff
@@ -68,85 +64,97 @@ describe('<TrackPanelBookmarks />', () => {
     changeDrawerViewAndOpen
   };
 
-  let wrapper: any;
-
   beforeEach(() => {
-    wrapper = mount(<TrackPanelBookmarks {...props} />);
     jest.resetAllMocks();
   });
 
-  it('renders without any error', () => {
-    expect(wrapper).toHaveLength(1);
+  it('renders previously viewed links', () => {
+    const { container } = render(<TrackPanelBookmarks {...props} />);
+    const links = [...container.querySelectorAll('.link')] as HTMLElement[];
+    const linkTexts = props.previouslyViewedObjects.map(({ label }) => label);
+
+    expect(
+      linkTexts.every((text) =>
+        links.find((link) => {
+          return link.innerHTML === text;
+        })
+      )
+    ).toBe(true);
   });
 
-  it('renders correct number of previously viewed links', () => {
-    const previouslyViewedLinksWrapper = wrapper.find(PreviouslyViewedLinks);
-
-    expect(previouslyViewedLinksWrapper.find('.link').length).toBe(
-      numberOfPreviouslyViewedObjects
+  it('closes the bookmarks modal when a bookmark link is clicked', () => {
+    render(<TrackPanelBookmarks {...props} />);
+    const previouslyViewedLinksContainer = screen.getByText('Previously viewed')
```

Comment:
Is this approach better than adding an ID to the container element? 
This test is more likely to break if we introduce a different element after the header.

Reply:
This is a very good point. I've been thinking about starting using data-test-ids for such cases, which the testing-library community recommends ([link](https://testing-library.com/docs/dom-testing-library/api-queries#bytestid)). I didn't want to add CSS selectors, such as ids or classes, specifically for testing purposes, but there is a babel plugin that can remove unnecessary html attributes, such as these test ids, from the production build, so that they do not get shipped to the end user (feels a bit dirty if they do). I've been a bit apprehensive of adding the extra babel plugin that removes these ids from the final build (or maybe just lazy), but now I guess is a good time to set this up.

Reply:
✔️

Reply:
Looks good 👍 
Should we use snake casing for the test IDs or it doesn't matter? It feels a little odd to use text with spaces in an ID field.